src/xbot/common/properties/PermanentStorageBase.java

```diff
@@ -1,24 +1,80 @@
 package xbot.common.properties;
 
 import java.sql.Connection;
+import java.sql.DatabaseMetaData;
 import java.sql.DriverManager;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
+
 import org.apache.log4j.Logger;
 
 /**
  *
  * @author John
  */
-public class PermanentStorage extends PermanentStorageProxy {
+public class PermanentStorageBase extends PermanentStorageProxy {
 
-    private static Logger log = Logger.getLogger(PermanentStorage.class);
+    private static Logger log = Logger.getLogger(PermanentStorageBase.class);
     
-    private String dbUrl = "jdbc:derby:/488Database;create=true";
+    private String dbUrlPreFormat = "jdbc:derby:%1s;create=true";
+    private String dbUrl = "";
     
-    public PermanentStorage() {
+    public PermanentStorageBase(String databaseDirectory) {
         super();
+        
+        dbUrl = String.format(dbUrlPreFormat, databaseDirectory);
+    }
+    
+    private boolean propertiesTableExists(Connection conn)
+    {
+		try {
+			DatabaseMetaData md = conn.getMetaData();
+            ResultSet tables = md.getTables(null, null, "PROPERTIES", null);
+            
+            if (tables.next())
+            {
+            	return true;
+            }
+            return false;
+	        
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		return false;
+    }
+    
+    public boolean obliterateStorage() {
```

Comment:
Might be one of those places where a javadoc comment explaining when it returns true vs false would be helpful